pkg/query/endpointset.go

```diff
@@ -76,17 +77,25 @@ func (es *grpcEndpointSpec) Addr() string {
 // Metadata method for gRPC endpoint tries to call InfoAPI exposed by Thanos components until context timeout. If we are unable to get metadata after
 // that time, we assume that the host is unhealthy and return error.
 func (es *grpcEndpointSpec) Metadata(ctx context.Context, client *endpointClients) (*endpointMetadata, error) {
-	resp, err := client.info.Info(ctx, &infopb.InfoRequest{}, grpc.WaitForReady(true))
-	if err != nil {
-		// Call Info method of StoreAPI, this way querier will be able to discovery old components not exposing InfoAPI.
-		metadata, merr := es.getMetadataUsingStoreAPI(ctx, client.store)
-		if merr != nil {
-			return nil, errors.Wrapf(merr, "fallback fetching info from %s after err: %v", es.addr, err)
+	if client.info != nil {
```

Comment:
I would add the same comment here otherwise readers will be surprised. OR just remove the whole `client.info`

Reply:
(We can get it back from git)

Reply:
You're right, without having the context it's surprising. I'll add the same comment to keep it consistent.